polygon/sync/tip_events.go

```diff
@@ -201,6 +203,16 @@ func (te *TipEvents) Run(ctx context.Context) error {
 	}, heimdall.WithEventsLimit(5))
 	defer milestoneObserverCancel()
 
+	metricsMilestoneObserverCancel := te.heimdallObserverRegistrar.RegisterMilestoneObserver(func(milestone *heimdall.Milestone) {
```

Comment:
would suggest to move these inside the `heimdall` package, can be registered as background daemons of the `heimdall.Service` inside its `Run` method

Reply:
moved to heimdall